source/screens/EepromScreen.cpp

```diff
@@ -0,0 +1,119 @@
+#include "EepromScreen.hpp"
+#include "Gfx.hpp"
+#include "ProcUI.hpp"
+#include "Utils.hpp"
+#include <cstdio>
+
+#include <coreinit/mcp.h>
+#include <coreinit/thread.h>
+#include <nsysccr/cdc.h>
+#include <nsysccr/cfg.h>
+#include <nn/ccr.h>
+
+namespace {
+
+bool WriteFile(const uint8_t *source, const std::string& dstPath)
+{
+    FILE* outf = fopen(dstPath.c_str(), "wb");
+    if (!outf) {
+        return false;
+    }
+        if (fwrite(source, 1, 4096, outf) != 0) { // DRC EEPROM size up until "DEADBABE" string appears to be 5E0h bytes, unless you dump the 2nd DRC. What the hell does it add..?
```

Comment:
You will write OOB data to your file if you always write 4096 bytes, but only pass a 772 byte buffer to the function.
Ideally you should pass a `std::span` or a size parameter to the function.

Reply:
I can fix this, but results for the auxiliary DRC will probably be inconsistent with the primary. I'll check though, but in a week.

Reply:
Okay I've checked my previous dumps. I'll hardcode the size to 304h bytes.
I'm not exactly sure why dumping EEPROM for DRC1 appends the EEPROM for DRC0 after the required data, but that should not matter.